Unify DPL documentation (#2285)
telerik/blazor-docs: integrations/document-processing-libraries.md

---
title: Telerik Document Processing  Libraries 
page_title: Document Processing Libraries - Integration with Telerik Products
description: "Get started with Telerik UI for {{ site.framework }} and learn how to work with the PdfProcessing, SpreadStreamProcessing, WordsProcessing, and ZipLibrary libraries as part of the Telerik Document Processing Libraries."
slug: dpl-in-blazor
position: 11
previous_url: /components/document-processing/overview, /components/document-processing/pdf, /components/document-processing/spreadsheet, /components/document-processing/spreadstream, /components/document-processing/words, /components/document-processing/zip
---

# Telerik Document Processing Libraries

> [Telerik Document Processing Libraries](https://docs.telerik.com/devtools/document-processing/introduction) is a bundle of UI-independent cross-platform libraries that enable you to process the most commonly used flow, fixed and spreadsheet document formats for web and desktop. The Document Processing library allows you to create, import, modify and export documents in the [supported formats](https://docs.telerik.com/devtools/document-processing/introduction#supported-formats) without relying on external dependencies like Adobe Acrobat or Microsoft Office:

Reviewer: ```suggestion
[Telerik Document Processing Libraries](https://docs.telerik.com/devtools/document-processing/introduction) is a bundle of UI-independent cross-platform libraries that enable you to process the most commonly used flow, fixed, and spreadsheet document formats for web and desktop. The Document Processing library allows you to create, import, modify, and export documents in the [supported formats](https://docs.telerik.com/devtools/document-processing/introduction#supported-formats) without relying on external dependencies like Adobe Acrobat or Microsoft Office.
```

![Telerik Document Processing  Libraries ](images/Telerik_Document_Processing_Libraries.png) 

|Library|Description|
|----|----|
| [RadPdfProcessing](https://docs.telerik.com/devtools/document-processing/libraries/radpdfprocessing/overview)|A processing library that allows you to create, import, and export PDF documents from your code. You can use it in any web or desktop .NET application without relying on third-party software like Adobe Acrobat.|

> |[RadSpreadProcessing](https://docs.telerik.com/devtools/document-processing/libraries/radspreadprocessing/overview)|A powerful library that enables you to create applications with native support for spreadsheet documents. With RadSpreadProcessing, you can create spreadsheets from scratch, modify existing documents or convert between the most common spreadsheet formats. You can save the generated workbook to a local file, stream, or stream it to the client browser.|

Reviewer: ```suggestion
|[RadSpreadProcessing](https://docs.telerik.com/devtools/document-processing/libraries/radspreadprocessing/overview)|A powerful library that enables you to create applications with native support for spreadsheet documents. With RadSpreadProcessing, you can create spreadsheets from scratch, modify existing documents, or convert between the most common spreadsheet formats. You can save the generated workbook to a local file, stream, or stream it to the client browser.|
```

> |[RadSpreadStreamProcessing](https://docs.telerik.com/devtools/document-processing/libraries/radspreadstreamprocessing/overview)|Spread streaming is a document processing paradigm that allows you to create or read big spreadsheet documents with great performance and minimal memory footprint. The key for the memory efficiency is that the spread streaming library writes the spreadsheet content directly to a stream without creating and preserving the spreadsheet document model in memory.|

Reviewer: ```suggestion
|[RadSpreadStreamProcessing](https://docs.telerik.com/devtools/document-processing/libraries/radspreadstreamprocessing/overview)|Spread streaming is a document processing paradigm that allows you to create or read big spreadsheet documents with great performance and minimal memory footprint. The key to memory efficiency is that the spread streaming library writes the spreadsheet content directly to a stream without creating and preserving the spreadsheet document model in memory.|
```

> |[RadWordsProcessing](https://docs.telerik.com/devtools/document-processing/libraries/radwordsprocessing/overview)|A processing library that allows you to create, modify and export documents to a variety of formats. Through the API, you can access each element in the document and modify, remove it or add a new one. The generated content you can save as a stream, as a file, or sent it to the client browser.|

Reviewer: ```suggestion
|[RadWordsProcessing](https://docs.telerik.com/devtools/document-processing/libraries/radwordsprocessing/overview)|A processing library that allows you to create, modify, and export documents to a variety of formats. Through the API, you can access, modify, remove, or add elements to the document. You can save the generated content as a stream or file, or send it to the client browser.|
```

|[RadZipLibrary](https://docs.telerik.com/devtools/document-processing/libraries/radziplibrary/overview)| It allows you to compress and combine files in ZIP archives, browse and extract files from existing ZIP archives and compress streams for easy file shipping and reduced storage space.|

## Available Versions

> Telerik Document Processing  Libraries are available in **.NET Framework** and **.NET Standard (.NET Core)** compatible versions. Both versions are [available as NuGet packages](https://docs.telerik.com/devtools/document-processing/getting-started/installation/nuget-packages). Read more in the *[Getting Started with Telerik Document Processing Libraries](https://docs.telerik.com/devtools/document-processing/getting-started/getting-started)* article.

Reviewer: ```suggestion
Telerik Document Processing  Libraries are available in .NET Framework and .NET Standard (.NET Core) compatible versions. Both versions are [available as NuGet packages](https://docs.telerik.com/devtools/document-processing/getting-started/installation/nuget-packages). Read more in the [Getting Started with Telerik Document Processing Libraries](https://docs.telerik.com/devtools/document-processing/getting-started/getting-started) article.
```
